tlslite/tlsrecordlayer.py

```diff
@@ -722,6 +722,19 @@ def _getMsg(self, expectedType, secondaryType=None, constructorType=None):
                                 yield result
                             continue
 
+                    # If we received a heartbeat request ...
+                    if recordHeader.type == ContentType.heartbeat:
```

Comment:
if that type was received but the extension was not negotiated, it should cause connection abort through `unexpected_message`